update guide of SYCL backend (#5254)
ggml-org/llama.cpp: README-sycl.md

|Intel built-in Arc GPU| Support| built-in Arc GPU in Meteor Lake|
|Intel iGPU| Support| iGPU in i5-1250P, i7-1165G7|

> Note: If the EUs (Execution Unit) in iGPU is less than 80, the inference speed will too slow to use.

Reviewer: ```suggestion
Note: If the EUs (Execution Unit) in iGPU is less than 80, the inference speed will be too slow to use.
```

Author: Yes, update

### Memory

The memory is a limitation to run LLM on GPUs.

> When run llama.cpp, there is print log to show the applied memory on GPU. You could know how many memory to be used in your case. Like `llm_load_tensors:            buffer size =  3577.56 MiB`.

Reviewer: ```suggestion
When run llama.cpp, there is print log to show the applied memory on GPU. You could know how much memory to be used in your case. Like `llm_load_tensors:            buffer size =  3577.56 MiB`.
```

Author: yes

sycl-ls
```

> There should be one or more level-zero devices. Please confirm that at least one GPU be present, like **[ext_oneapi_level_zero:gpu:0]**.

Reviewer: ```suggestion
There should be one or more level-zero devices. Please confirm that at least one GPU is present, like **[ext_oneapi_level_zero:gpu:0]**.
```

Author: yes

sycl-ls
```

> There should be one or more level-zero devices. Please confirm that at least one GPU be present, like **[ext_oneapi_level_zero:gpu:0]**.

Author: yes

[opencl:cpu:1] Intel(R) OpenCL, 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz OpenCL 3.0 (Build 0) [2023.16.10.0.17_160000]
[opencl:gpu:2] Intel(R) OpenCL Graphics, Intel(R) Iris(R) Xe Graphics OpenCL 3.0 NEO  [31.0.101.5186]
[ext_oneapi_level_zero:gpu:0] Intel(R) Level-Zero, Intel(R) Iris(R) Xe Graphics 1.3 [1.3.28044]
```

3. Install cmake & make

a. Download & install cmake for Windows: https://cmake.org/download/

b. Download & install make for Windows provided by mingw-w64

> - Download binary package for Windows in https://github.com/niXman/mingw-builds-binaries/releases.

Reviewer 2: VS build is more common for Windows users. Just a thought.

Author: Yes. But I meet this issue in a new PC.
I think it's better record here.

Reviewer 3: We recommend w64devkit for MinGW in the main README, I don't know why we would recommend yet another MinGW distribution here.

Author: Here use the make.exe in MinGW only.
w64devkit is docker env.
In the chapter **Windows**, we provide the native build method. So not use w64devkit.
In the chapter **Docker**, there is docker method.

I will find another method to install make.exe without MinGW.

Thank you!

Author: @cebtenzzre Update as your suggestion: https://github.com/ggerganov/llama.cpp/pull/5297

  Miss to enable oneAPI running environment.

- Meet compile error.

  Remove folder **build** and try again.

- I can **not** see **[ext_oneapi_level_zero:gpu:0]** afer install GPU driver in Linux.

  Please run **sudo sycl-ls**.

>   If see it in result, please add video/render group to your ID:

Reviewer: ```suggestion
  If you see it in result, please add video/render group to your ID:
```

Author: yes

  ```
  sudo usermod -aG render username
  sudo usermod -aG video username
  ```

  Then **relogin**.

>   If no see it, please check the installation GPU steps again.

Reviewer: ```suggestion
  If you do not see it, please check the installation GPU steps again.
```

Author: yes
